pages/set/_.vue

```diff
@@ -128,17 +151,31 @@
     // TODO: error handling for Nuxt 2.12 fetch()
     //       https://nuxtjs.org/blog/understanding-how-fetch-works-in-nuxt-2-12/#error-handling
     async fetch() {
-      const set = await this.$sets.getSet(this.$route.params.pathMatch, {
-        profile: 'itemDescriptions'
-      });
-
-      this.id = set.id;
-      this.title = set.title;
-      this.description = set.description;
-      this.visibility = set.visibility;
-      this.creator = set.creator;
-      this.total = set.total || 0;
-      this.items = set.items;
+      const token = this.$auth.loggedIn ? this.$auth.getToken('keycloak') : null;
+      const headers = {};
+      if (token) headers['Authorization'] = token;
+      const apiUrl = `${this.apiConfig.set.origin}${this.apiConfig.set.path}/${this.$route.params.pathMatch}?wskey=${this.apiConfig.set.key}&profile=itemDescriptions`;
+      const set = await fetch(apiUrl, { headers })
```

Comment:
Why is all of this here, duplicating the functionality from the set plugin?

If it's just to get the status code, then apiError includes that as a property named `statusCode`, so just catch the error.